boards/adafruit/qt_py_esp32s3/board.yml

```diff
@@ -0,0 +1,11 @@
+board:
+  name: adafruit_qt_py_esp32s3
+  full_name: QT Py ESP32-S3
+  vendor: adafruit
+  socs:
+  - name: esp32s3
+    variants:
+    - name: 'psram'
+      cpucluster: 'procpu'
+    - name: 'psram'
+      cpucluster: 'appcpu'
```

Comment:
if this comes in 2 designs then it should be using hardware versions not variants unless you mean there is only 1 board and you can pick if you use the psram or not

Reply:
Adafruit offers 2 boards with the same name, with the only difference being that one has "with PSRAM" in the name. The best I can tell, they use the same schematic and use a BOM swap to have the different SOCs placed in manufacturing.

Both products are still in active production:
- [Adafruit QT Py ESP32-S3 WiFi Dev Board with STEMMA QT - 8 MB Flash / No PSRAM](https://www.adafruit.com/product/5426)
- [Adafruit QT Py S3 with 2MB PSRAM WiFi Dev Board with STEMMA QT](https://www.adafruit.com/product/5700)

Reply:
Use board revisions: https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#multiple-board-revisions

Reply:
Except the boards aren't revisions of each other, they are distinct products and don't fall into a semantic versioning scheme. Unless you're trying to suggest that using the amount of flash a board has should be using the revision scheme?

Reply:
If they're distinct products then why would they be variants?

Reply:
The only difference between the two variants is that one includes in their device tree

```
#include <espressif/esp32s3/esp32s3_wroom_n8.dtsi>
```

while the other includes

```
#include <espressif/esp32s3/esp32s3_wroom_n4r2.dtsi>
```

The other way to accomplish this would be to have 2 complete board directories with near identical contents (including docs), and that seems a bit overkill for what is a 1 line change. I felt that the most effective way to support both versions of the board is to use the variant system.

It gets even more confusing with the physical boards too, as the only way to visually tell the boards apart is to look at the part number on the SOC itself.

Reply:
Having 2 different boards just because SPIRAM sounds odd, I agree about that. 
At the same time, It sounds those boards have no "revision" number between those 2 parts.

The 3rd option and simpler approach would be to only include the board without PSRAM and let application handling additional .conf/.overlay files to include PSRAM entries. You can add this info into documentation describing how to do so..